common/functions/src/scalars/others/inet_aton.rs

```diff
@@ -13,80 +13,164 @@
+    pub fn desc() -> Function2Description {
+        Function2Description::creator(Box::new(Self::try_create))
             .features(FunctionFeatures::default().deterministic().num_arguments(1))
     }
 }
 
-impl Function for InetAtonFunction {
+impl<const SUPPRESS_PARSE_ERROR: bool> Function2 for InetAtonFunctionImpl<SUPPRESS_PARSE_ERROR> {
     fn name(&self) -> &str {
         &*self.display_name
     }
 
-    fn return_type(&self, args: &[DataTypeAndNullable]) -> Result<DataTypeAndNullable> {
-        let data_type = if args[0].is_string() || args[0].is_null() {
-            Ok(DataType::UInt32)
-        } else {
-            Err(ErrorCode::IllegalDataType(format!(
+    fn return_type(&self, args: &[&DataTypePtr]) -> Result<DataTypePtr> {
+        let input_type = remove_nullable(args[0]);
+
+        let output_type = match input_type.data_type_id() {
+            TypeID::String | TypeID::Null => Ok(type_primitive::UInt32Type::arc()),
+            _ => Err(ErrorCode::IllegalDataType(format!(
                 "Expected string or null type, but got {}",
-                args[0]
-            )))
+                args[0].name()
+            ))),
         }?;
-        // For invalid input, the function returns null, so the nullable is true.
-        Ok(DataTypeAndNullable::create(&data_type, true))
+
+        if SUPPRESS_PARSE_ERROR {
+            // For invalid input, we suppress parse error and return null. So the return type must be nullable.
+            return Ok(Arc::new(NullableType::create(output_type)));
+        }
+
+        if args[0].is_nullable() {
+            Ok(Arc::new(NullableType::create(output_type)))
+        } else {
+            Ok(output_type)
+        }
+    }
+
+    fn eval(&self, columns: &ColumnsWithField, input_rows: usize) -> Result<ColumnRef> {
+        if SUPPRESS_PARSE_ERROR {
+            let viewer = ColumnViewer::<Vec<u8>>::create(columns[0].column())?;
+
+            let mut builder = NullableColumnBuilder::<u32>::with_capacity(input_rows);
+
+            for i in 0..input_rows {
+                // We skip the null check because the function has passthrough_null is true.
+                // This is arguably correct because the address parsing is not optimized by SIMD, not quite sure how much we can gain from skipping branch prediction.
+                // Think about the case if we have 1000 rows and 999 are Nulls.
+                let input = viewer.value(i);
+                let addr_str = String::from_utf8_lossy(input);
+                match addr_str.parse::<Ipv4Addr>() {
+                    Ok(addr) => {
+                        let addr_binary: u32 = u32::from(addr);
+                        builder.append(addr_binary, viewer.valid_at(i));
+                    }
+                    Err(_) => builder.append_null(),
+                }
+            }
+            return Ok(builder.build(input_rows));
+        }
+
+        if columns[0].column().data_type_id() == TypeID::Null {
```

Comment:
It looks we already did it in the adaptor.

Reply:
I have passthrough is false for this case ...... 😢